### ISSUE TITLE
关于梯度none问题

### ISSUE THREAD

**Issue by feanuolian** [GeWu-Lab/InfoReg_CVPR2025#10]
In main.py at line 504, the code `fim[name] += (module.weight.grad * module.weight.grad)` is encountering a NoneType error during computation. Could you please explain why this might be happening?

**Comment by celestial1122** [comment]
The error stems from the fact that our AVClassifier contains auxiliary heads (head_audio, head_video) and gated layers that aren't always active. In our current training logic, we calculate unimodal losses by slicing the shared head weights, meaning the specific head_audio and head_video modules never receive gradients.  Adding a simple is not None check ensures we only compute the Fisher Information for active layers.
So, you can add:
`if module.weight.requires_grad and module.weight.grad is not None:`

**Comment by bissotow** [comment]
The code is mentioned on line 16 of the main function
from utils.utils import setup_seed, weight_init
However, the uls package was not seen in the file

**Comment by feanuolian** [comment]
Thank you so much for your prompt and detailed response to my question!
